auth/authenticators/iam-assume-authenticator.ts

```diff
@@ -0,0 +1,100 @@
+/**
+ * (C) Copyright IBM Corp. 2024.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Authenticator } from './authenticator';
+import { IamAssumeTokenManager } from '../token-managers';
+import { IamRequestOptions, IamRequestBasedAuthenticator } from './iam-request-based-authenticator';
+
+/** Configuration options for IAM Assume authentication. */
+export interface Options extends IamRequestOptions {
+  /** The IAM api key */
+  apikey: string;
+
+  /**
+   * Specify exactly one of [iamProfileId, iamProfileCrn, or iamProfileName] to
+   * identify the trusted profile whose identity should be used. If iamProfileId
+   * or iamProfileCrn is used, the trusted profile must exist in the same account.
+   * If and only if IAMProfileName is used, then iamAccountId must also be
```

Comment:
```suggestion
   * If and only if iamProfileName is used, then iamAccountId must also be
```
Just to be consistent with the rest. :)